embassy-nrf/src/uarte.rs

```diff
@@ -601,17 +662,25 @@ impl<'d, T: Instance> UarteRx<'d, T> {
         r.rxd.maxcnt.write(|w| unsafe { w.maxcnt().bits(len as _) });
 
         r.events_endrx.reset();
-        r.intenset.write(|w| w.endrx().set());
+        r.events_error.reset();
+        r.intenset.write(|w| {
+            w.endrx().set();
+            w.error().set()
+        });
 
         compiler_fence(Ordering::SeqCst);
 
         trace!("startrx");
         r.tasks_startrx.write(|w| unsafe { w.bits(1) });
 
-        poll_fn(|cx| {
+        let result = poll_fn(|cx| {
             s.endrx_waker.register(cx.waker());
+
+            if let Err(e) = self.check_and_clear_errors() {
+                return Poll::Ready(Err(e));
```

Comment:
This needs to stop the DMA transfer, otherwise DMA will keep writing to the buffer after returning, which is UB.